linux/LibSensors.c

```diff
@@ -120,26 +120,29 @@ int LibSensors_reload(void) {
    return sym_sensors_init(NULL);
 }
 
-static int tempDriverPriority(const sensors_chip_name* chip) {
-   static const struct TempDriverDefs {
+typedef enum TempDriver_ {
+   TD_CORETEMP,
+   TD_CPUTEMP,
+   TD_CPUTHERMAL,
+   TD_K10TEMP,
+   TD_ZENPOWER,
+   TD_ACPITZ,
+   TD_UNKNOWN,
+} TempDriver;
+
+static const struct TempDriverDefs {
       const char* prefix;
       int priority;
-   } tempDrivers[] =  {
-      { "coretemp",    0 },
-      { "via_cputemp", 0 },
-      { "cpu_thermal", 0 },
-      { "k10temp",     0 },
-      { "zenpower",    0 },
-      /* Low priority drivers */
-      { "acpitz",      1 },
-   };
-
-   for (size_t i = 0; i < ARRAYSIZE(tempDrivers); i++)
-      if (String_eq(chip->prefix, tempDrivers[i].prefix))
-         return tempDrivers[i].priority;
-
-   return -1;
-}
+      TempDriver driver;
+} tempDrivers[] =  {
+   { "coretemp",    0, TD_CORETEMP },
+   { "via_cputemp", 0, TD_CPUTEMP },
+   { "cpu_thermal", 0, TD_CPUTHERMAL },
+   { "k10temp",     0, TD_K10TEMP },
+   { "zenpower",    0, TD_ZENPOWER },
+   /* Low priority drivers */
+   { "acpitz",      1, TD_ACPITZ },
```

Comment:
Is there a possibility that a temperature driver comes with two names?
Because I wonder if the table can be simplified to this:

```c
   [TD_CORETEMP] = { "coretemp", 0 },
   [TD_CPUTEMP] = { "via_cputemp", 0 },
   [TD_CPUTHERMAL] = { "cpu_thermal", 0 },
   [TD_K10TEMP] = { "k10temp", 0 },
   [TD_ZENPOWER] = { "zenpower", 0 },
   /* Low priority drivers */
   [TD_ACPITZ] = { "acpitz", 1 },
```